src/stark/stark_commit.cairo

```diff
@@ -1 +1,114 @@
+    oods::verify_oods,
+};
 
+#[derive(Drop, Copy)]
+struct InteractionValuesAfterTraces {
+    // n_constraints Coefficients for the AIR constraints.
+    coefficients: Span<felt252>,
+}
+
+#[derive(Drop, Copy)]
+struct InteractionValuesAfterOods {
+    // n_oods_values coefficients for the boundary polynomial validating the OODS values.
+    coefficients: Span<felt252>,
+}
+
+// STARK commitment phase.
+fn stark_commit(
+    ref channel: Channel,
+    public_input: @PublicInput,
+    unsent_commitment: @StarkUnsentCommitment,
+    config: @StarkConfig,
+    stark_domains: @StarkDomains,
+) -> StarkCommitment {
+    let traces_commitment = traces_commit(
+        ref channel,
+        N_INTERACTION_ELEMENTS,
+        public_input,
+        *unsent_commitment.traces,
+        *config.traces,
+    );
+
+    let composition_alpha = channel.random_felt_to_prover();
+    let traces_coefficients = powers_array(
+        1, composition_alpha, N_CONSTRAINTS.try_into().unwrap(),
+    );
+
+    let interaction_after_traces = InteractionValuesAfterTraces {
+        coefficients: traces_coefficients.span(),
+    };
+
+    let composition_commitment = table_commit(*unsent_commitment.composition, *config.composition,);
+
+    let interaction_after_composition = channel.random_felt_to_prover();
+
+    let n_oods_values = MASK_SIZE + CONSTRAINT_DEGREE.into();
+    channel.read_felts_from_prover(*unsent_commitment.oods_values);
+
+    let interaction_elements = InteractionElements {
+        memory_multi_column_perm_perm_interaction_elm: 0,
+        memory_multi_column_perm_hash_interaction_elm0: 0,
+        rc16_perm_interaction_elm: 0,
+        diluted_check_permutation_interaction_elm: 0,
+        diluted_check_interaction_z: 0,
+        diluted_check_interaction_alpha: 0,
+    };
+
+    let oods = *unsent_commitment.oods_values;
+    let mut mask_values = array![];
+    let mut i = 0;
+    loop {
+        if i == oods.len() - 2 {
+            break;
+        }
+
+        mask_values.append(*oods.at(i));
+
+        i += 1;
+    };
+    let split_polynomials = array![*oods.at(i), *oods.at(i + 1)];
+    let oods_values = OodsValues { mask_values, split_polynomials };
+
+    verify_oods(
+        oods_values,
+        interaction_elements,
+        public_input,
+        traces_coefficients,
+        interaction_after_composition,
+        *stark_domains.trace_generator,
+        *stark_domains.trace_domain_size
+    );
+
+    let oods_alpha = channel.random_felt_to_prover();
+    let oods_coefficients = powers_array(1, oods_alpha, n_oods_values.try_into().unwrap());
+    let interaction_after_oods = InteractionValuesAfterOods {
+        coefficients: oods_coefficients.span()
+    };
+
+    let fri_commitment = fri_commit(ref channel, *unsent_commitment.fri, *config.fri);
+
+    proof_of_work_commit(ref channel, *unsent_commitment.proof_of_work, *config.proof_of_work);
+
+    StarkCommitment {
```

Comment:
I've converted `interaction_after_oods` to `Array` but let's leave `oods_values` as `Span` because I would need to convert it manually to `Array` in the function body because it's provided as a snapshot function argument.